storage/s3/s3.go

```diff
@@ -163,23 +170,20 @@ func (s *s3Storage) List(p string) ([]storage.FileEntry, error) {
 		"key":    key,
 	}).Info("finding objects")
 
-	exists, err := s.client.BucketExists(bucket)
+	exists, err := s.client.BucketExists(s.ctx, bucket)
 	if err != nil {
 		return nil, fmt.Errorf("error when accessing bucket %s: %w", bucket, err)
 	} else if !exists {
 		return nil, fmt.Errorf("bucket %s does not exist", bucket)
 	}
 
-	// Create a done channel to control 'ListObjectsV2' go routine.
-	doneCh := make(chan struct{})
-
-	// Indicate to our routine to exit cleanly upon return.
-	defer close(doneCh)
-
 	var objects []storage.FileEntry
-	isRecursive := true
-	objectCh := s.client.ListObjectsV2(bucket, key, isRecursive, doneCh)
-	for object := range objectCh {
+	opts := minio.ListObjectsOptions{
+		Recursive: true,
+		Prefix:    key,
+	}
+
+	for object := range s.client.ListObjects(s.ctx, bucket, opts) {
```

Comment:
Why are you switching from `ListObjectsV2` to `ListObjects`?

Reply:
They got rid of that in v7

https://docs.min.io/docs/golang-client-api-reference.html#ListObjects

Reply:
ah, thanks for pointing that out :)